### DISCUSSION TITLE
layoutData in lazy loading not working

### DISCUSSION THREAD

**Discussion by restu-lomboe** [livewire/livewire#9089] [1. Help]
### Livewire version

v3.0

### Laravel version

v10.10

### Which PHP version are you using?

PHP 8.3

### What is the problem?

layoutData for meta title not working when use lazy loading,
by default i use laravel template layout for wire navigate 
`php artisan livewire:layout`

### Code snippets

```php
<?php
namespace App\Livewire;

use Livewire\Component;
use Livewire\Attributes\Lazy;

#[Lazy]
class Simple extends Component
{
    public $course_title;
    public $course_description;
    public $course_image;
    
    public function mount(){

        $this->course_title = 'this is title';
        $this->course_description = 'this is description';
        $this->course_image = 'this is image';
    }
    
    public function placeholder(){
        return view('livewire.skeleton');
    }
    
    public function render()
    {
        return view('livewire.simple')->layoutData([
            'title' => $this->course_title,
            'description' => $this->course_description,
            'image' => $this->course_image,
        ]);
    }
}
```

```blade skeleton
<div>
    <div class="px-6 md:px-20 max-sm:px-8 max-phone:px-4 pb-52 animate-pulse">
        <div class="max-w-[85rem] lg:max-w-full py-10 lg:py-14 mx-auto">
            <div class="flex flex-col h-full lg:flex-row">
                <div
                    class="w-full lg:w-1/4 order-2 lg:order-1 border bg-white rounded-xl h-52 lg:h-[600px] 2xl:h-[700px]">
                    <ul class="mt-5 space-y-3 p-3">
                        <li class="w-full h-4 bg-gray-200 rounded-full dark:bg-neutral-700"></li>
                        <li class="w-full h-4 bg-gray-200 rounded-full dark:bg-neutral-700"></li>
                        <li class="w-full h-4 bg-gray-200 rounded-full dark:bg-neutral-700"></li>
                        <li class="w-full h-4 bg-gray-200 rounded-full dark:bg-neutral-700"></li>
                    </ul>
                </div>
                <div class="w-full lg:w-3/4 mr-8 rounded-xl bg-gray-200 border border-gray-200"></div>
            </div>
        </div>
        <p class="h-4 bg-gray-200 rounded-full dark:bg-neutral-700" style="width: 40%;"></p>
        <ul class="mt-5 space-y-3">
            <li class="w-full h-4 bg-gray-200 rounded-full dark:bg-neutral-700"></li>
            <li class="w-full h-4 bg-gray-200 rounded-full dark:bg-neutral-700"></li>
            <li class="w-full h-4 bg-gray-200 rounded-full dark:bg-neutral-700"></li>
            <li class="w-full h-4 bg-gray-200 rounded-full dark:bg-neutral-700"></li>
        </ul>
    </div>
</div>
```

### Wirebox URL

https://wirebox.app/b/x65jd

### How do you expect it to work?

how to make dinamis layoutData when use lazy loading

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my help request will be removed if I haven't met the criteria above.

**Comment by danie-ramdhani** [comment]
yeah, it's not working (or maybe limitation) because the layout is rendered first then the livewire render the component when the element is observed.

you need to use javascript if you want to make it dynamically in layout view:

layout view:

```html
<!-- ... -->
<div id="foo"></div>
<!-- ... -->
```

lazy component:

```php
public string $foo = 'bar';

public function render()
{
    $this->js(<<<'JS'
        document.querySelector('#foo').innerText = $wire.foo
    JS);
}
```

**Comment by restu-lomboe** [comment]
still can't, because by default if you use lazy loading, meta tags it will only read the rendered data the first time when you share link, so when you update the meta tags after the data process is complete the meta tags change but are not read when we share the link

**Comment by joshhanley** [comment]
@restu-lomboe you can't update layout data from a lazy component (or any subsequent requests in any component) as the layout has already been rendered on the screen in the initial request.

The only way to update the layout will be to use javascript, or make sure metadata is loaded on page load and use `wire:navigate` when changing courses.
